File: №1_4.go
package main

import (
	"fmt"
	"math"
)

// Проверка на простоту
func isPrime(n int) bool {
	if n < 2 {
		return false
	}
	for i := 2; i <= int(math.Sqrt(float64(n))); i++ {
		if n%i == 0 {
			return false
		}
	}
	return true
}

func main() {
	var N int
	fmt.Scan(&N)

	distance := 0

	for {
		if isPrime(N-distance) || isPrime(N+distance) {
			fmt.Println(distance)
			break
		}
		distance++
	}
}

File: №1_4.cpp
#include <iostream>
#include <cmath>
using namespace std;

bool isPrime(int n) {
    if (n < 2) return false;
    for (int i = 2; i <= sqrt(n); i++) {
        if (n % i == 0) return false;
    }
    return true;
}

int main() {
    int N;
    cin >> N;

    int distance = 0;

    while (true) {
        if (isPrime(N - distance) or isPrime(N + distance)) {
            cout << distance << endl;
            break;
        }
        distance++;
    }

    return 0;
}